### PR TITLE
feat: cmd.exe detection heuristic

### PR DESCRIPTION
**Pull request by samypr100**
## Summary

Follow up from discussion in https://github.com/astral-sh/uv/pull/2223

Detect CMD.exe by checking if `PROMPT` env var is set on windows, otherwise assume it's PowerShell.

Note, this will not work if user modifies their system env vars to include `PROMPT` by default or if they launch nested PowerShell from Command Prompt (e.g. `Developer PowerShell for VS 2022`).

## Test Plan

Only tested locally, although we try to add some CI tests that specifically use CMD.exe

Command Prompt
```
Microsoft Windows [Version 10.0.19044.3086]
(c) Microsoft Corporation. All rights reserved.

Z:\Users\samypr100\dev\uv>Z:\Users\samypr100\.cargo\bin\cargo.exe +stable run --color=always -- venv "Foo Bar"
    Finished dev [unoptimized + debuginfo] target(s) in 0.69s
     Running `target\debug\uv.exe venv "Foo Bar"`
Using Python 3.12.2 interpreter at: Z:\Users\samypr100\AppData\Local\Programs\Python\Python312\python.exe
Creating virtualenv at: Foo Bar
Activate with: "Foo Bar\Scripts\activate"
```

Power Shell
```
Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

Try the new cross-platform PowerShell https://aka.ms/pscore6

PS Z:\Users\samypr100\dev\uv>Z:\Users\samypr100\.cargo\bin\cargo.exe +stable run --color=always -- venv "Foo Bar"
    Finished dev [unoptimized + debuginfo] target(s) in 0.63s
     Running `target\debug\uv.exe venv "Foo Bar"`
Using Python 3.12.2 interpreter at: Z:\Users\samypr100\AppData\Local\Programs\Python\Python312\python.exe
Creating virtualenv at: Foo Bar
Activate with: & "Foo Bar\Scripts\activate"
```

### CODE REVIEW

File: crates/uv/src/commands/venv.rs [hunk 1]
```diff
@@ -255,14 +262,17 @@ fn shlex_posix(executable: impl AsRef<Path>) -> String {
 }
 
 /// Quote a path, if necessary, for safe use in `PowerShell`.
-fn shlex_windows(executable: impl AsRef<Path>) -> String {
+fn shlex_windows(executable: impl AsRef<Path>, shell: Shell) -> String {
```
Comment (charliermarsh) [inline crates/uv/src/commands/venv.rs:265]: Maybe we should just merge `shlex_posix` and `shlex_windows`, and pass in `Shell` to toggle the behavior -- wdyt?

Comment (samypr100) [reply crates/uv/src/commands/venv.rs:265]: I kinda like the separation between Posix and Windows, but I'm biased 😆

Comment (charliermarsh) [reply crates/uv/src/commands/venv.rs:265]: Sounds good :)

**Review by charliermarsh — APPROVED** [review]
Thanks!

**Pull request merged**